File: python/009/main.py
# ProjectEuler 009
# Python / main.py
# copyright (c) 2015 Susisu
#
# Python 3.4.2
# `python3 main.py`


def main():
    for a in range(1, 333):
        b = a
        while b < 1000 - a - b:
            if 2 * a * b - 2000 * (a + b) + 1000000 == 0:
                print(a * b * (1000 - a - b))
                return
            b += 1

main()

File: lua/009/main.lua
--[[
    ProjectEuler 009
    Lua / main.lua
    copyright (c) 2015 Susisu

    Lua 5.1.5
    `lua main.lua`
]]

function main()
    for a = 1, 332 do
        for b = a + 1, (1000 - a) / 2 do
            local c = 1000 - a - b
            if a * a + b * b == c * c then
                print(a * b * c)
                return
            end
        end
    end
end

main()

